cbioportal_etl/main.py

```diff
@@ -0,0 +1,277 @@
+    parser.add_argument("-ao", "--active-only", required=False, action="store_true", help="Only include active files")
+    parser.add_argument("-rm", "--rm-na", required=False, action="store_true", help="Remove entries where file_id and s3_path are NA")
+    parser.add_argument("-d", "--debug", required=False, action="store_true", help="Enable debug mode")
+    parser.add_argument("-o", "--overwrite", required=False, action="store_true", help="Overwrite files if they already exist")
+
+    # Arguments for Step 5 - genomics_file_cbio_package_build.py
+    parser.add_argument("-dpc", "--data-processing-config", required=False, help="Data processing configuration file. Default: value inputted for --study + '_data_processing_config.json'")
+    parser.add_argument("-sf", "--study-flag", required=True, choices=["both", "kf", "dgd"], help="Study flag")
+    parser.add_argument("-l", "--legacy", required=False, action="store_true", help="Enable legacy mode")
+
+    return parser
+
+def execute_step(step_number, args, steps):
+    step = steps.get(step_number)
+    if not step:
+        sys.stderr.write(f"Step {step_number} not defined in the pipeline.\n")
+        return
+
+    sys.stderr.write(f"\nRunning Step {step_number}: {step['description']}\n\n")
+
+    command = step["command"](args)
+    try:
+        subprocess.run(command, check=True)
+        sys.stderr.write(f"\nCompleted Step {step_number}\n")
+    except subprocess.CalledProcessError as e:
+        sys.stderr.write(f"\nError in Step {step_number}. Ensure all dependencies are satisfied and try again.\n")
+        sys.exit(1)
+            
+def main():
+    parser = setup_parser()
+    args = parser.parse_args()
+
+    if args.ref_dir is None:
+        args.ref_dir = REFS_DIR
+
+    # Resolve config file paths
+    default_meta_config = os.path.join(CONFIGS_DIR, f"{args.study}_case_meta_config.json")
+    default_data_config = os.path.join(CONFIGS_DIR, f"{args.study}_data_processing_config.json")
+    args.meta_config_file = resolve_config_path(args.meta_config_file, default_meta_config)
+    args.data_processing_config = resolve_config_path(args.data_processing_config, default_data_config)
+
+    # Download repo with validateData.py 
+    if "5" in args.steps or "all" in args.steps:
+        fetch_validator_scripts()
+
+    steps = {
+        1: {
+            "description": "Get study metadata",
+            "output_files": [
+                "datasheets/",
+                "cbio_file_name_id.txt"
+            ],
+            "command": lambda args: (
+                [
+                    "python3",
+                    os.path.join(SCRIPTS_DIR, "get_study_metadata.py"),
+                    "-d", args.db_ini,
+                    "-p", args.profile,
+                    "-c", args.meta_config_file,
+                    "-r", args.ref_dir
+                ]
+                + (["-a"] if args.all else [])
+            )
```

Comment:
I'm dubious about using subprocess to run python. I feel like its probably better to use `import`, but its up to you.